src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs

```diff
@@ -147,4 +148,17 @@ private static SyntaxNode DestructorDeclaration(this SyntaxGenerator syntaxGener
             return decl.ReplaceNode(decl.ReturnType, SyntaxFactory.IdentifierName(SyntaxFactory.Identifier(string.Empty)));
         }
     }
+
+    public static class SyntaxNodeExtensions
+    {
+        public static SyntaxNode Rewrite<T>(this SyntaxNode node, params object?[] p) where T : CSharpSyntaxRewriter
+        {
+            T? obj = (T?)Activator.CreateInstance(typeof(T), p);
```

Comment:
Why not just pass in the constructed object? So you wouldn't have `params object?[] p` either. You'd just have `obj` as a parameter that's already been constructed, and all you have to do is `Visit` it.

Reply:
The use of Activator.CreateInstance here seems superfluous. We'd typically prefer just using the ctor directly, e.g. instead of:
```C#
.Rewrite<TypeDeclarationSyntaxRewriter>()
```
doing
```C#
.Rewrite(new TypeDeclarationSyntaxRewriter(), node)
```
or something similar.

Reply:
The new constraint still uses Activator.CreateInstance. Why not just use the constructor rather than reflection?

Reply:
Updated